sixtyfps_runtime/rendering_backends/mcu/lib.rs

```diff
@@ -13,81 +13,205 @@ only be used with `version = "=x.y.z"` in Cargo.toml.
 */
 #![doc(html_logo_url = "https://sixtyfps.io/resources/logo.drawio.svg")]
 #![cfg_attr(not(feature = "simulator"), no_std)]
+#![cfg_attr(feature = "pico-st7789", feature(alloc_error_handler))]
 
 extern crate alloc;
 
-use alloc::boxed::Box;
-use alloc::rc::Rc;
-use alloc::string::String;
-use sixtyfps_corelib::{
-    graphics::{Image, Size},
-    window::Window,
-    ImageInner,
-};
-
 #[cfg(feature = "simulator")]
 mod simulator;
 
 #[cfg(feature = "simulator")]
-use simulator::*;
+use simulator::event_loop;
 
 mod renderer;
-pub struct Backend;
 
-impl sixtyfps_corelib::backend::Backend for Backend {
-    fn create_window(&'static self) -> Rc<Window> {
-        sixtyfps_corelib::window::Window::new(|window| SimulatorWindow::new(window))
-    }
+#[cfg(all(not(feature = "std"), feature = "unsafe_single_core"))]
+use sixtyfps_corelib::unsafe_single_core;
 
-    fn run_event_loop(&'static self, behavior: sixtyfps_corelib::backend::EventLoopQuitBehavior) {
-        simulator::event_loop::run(behavior);
-    }
+#[cfg(all(not(feature = "std"), feature = "unsafe_single_core"))]
+use sixtyfps_corelib::thread_local_ as thread_local;
 
-    fn quit_event_loop(&'static self) {
-        crate::event_loop::with_window_target(|event_loop| {
-            event_loop.event_loop_proxy().send_event(simulator::event_loop::CustomEvent::Exit).ok();
-        })
-    }
+#[cfg(not(feature = "simulator"))]
+// Backend to render one snapshot frame
+mod snapshotbackend {
```

Comment:
Would be better if this goes to another file.
But this refactoring can also be done later.
I guess this will go away when we have an event loop?
Otherwise should this be its own feature like "snapshot" or so?

Reply:
Ah, a feature is a neat idea. I can give that a try. Although I think I'd prefer to do that in a later refactoring.

The idea would be to get this change polished and in, rebase mcu on top where only printerdemo changes are.